eureka/src/test/java/com/linecorp/armeria/client/eureka/EurekaEndpointGroupTest.java

```diff
@@ -29,19 +29,26 @@
 import com.netflix.discovery.converters.wrappers.CodecWrappers;
 import com.netflix.discovery.converters.wrappers.CodecWrappers.JacksonJson;
 import com.netflix.discovery.converters.wrappers.EncoderWrapper;
+import com.netflix.discovery.shared.Application;
 import com.netflix.discovery.shared.Applications;
 import com.netflix.discovery.util.InstanceInfoGenerator;
 
 import com.linecorp.armeria.client.Endpoint;
 import com.linecorp.armeria.common.HttpResponse;
 import com.linecorp.armeria.common.HttpStatus;
 import com.linecorp.armeria.common.MediaType;
+import com.linecorp.armeria.common.annotation.Nullable;
+import com.linecorp.armeria.internal.common.eureka.InstanceInfo;
 import com.linecorp.armeria.server.ServerBuilder;
 import com.linecorp.armeria.testing.junit5.server.ServerExtension;
 
+import io.netty.util.AttributeKey;
+
 class EurekaEndpointGroupTest {
 
     private static final EncoderWrapper encoder = CodecWrappers.getEncoder(JacksonJson.class);
+    private static final String APP_WITH_METADATA = "with-metadata";
+    private static final AttributeKey<InstanceInfo> INSTANCE_INFO = AttributeKey.valueOf("instanceInfo");
```

Comment:
Most users might not know the magic string `instanceInfo`.
How about adding a utility that can return `InstanceInfo` from an `Endpoint`.
```java
public final class InstanceInfoUtil {
    @Nullable
    public InstanceInfo get(Endpoint endpoint) {
        return endpoint.attr(EurekaEndpointGroup.INSTANCE_INFO);
    }
}

Reply:
The InstanceInfo is still in the internal package. We should move them to a public package in this MR or a separate MR?

Reply:
Do you think there's a chance that a user wants to extract `InstanceInfo` from an `Endpoint`?

Reply:
We may want to create HealthCheckedEndpointGroup base on the InstanceInfo 'healthCheckUrl'.  But I am not sure really need it, We do not use HealthCheckContextGroup with EurekaEndpointGroup yet.

Reply:
If so, let's make `InstanceInfo` public when we really need it. 😄 
So, let's just use `AttributeKey<InstanceInfo> INSTANCE_INFO;` from `EurekaEndpointGroup` instead of introducing a util class. 😉

Reply:
> If so, let's make InstanceInfo public when we really need it. 😄
>   So, let's just use AttributeKey<InstanceInfo> INSTANCE_INFO; from EurekaEndpointGroup instead of introducing a util class. 😉

Did That mean we expose  1) `AttributeKey<InstanceInfo> INSTANCE_INFO` as a public static field but 2) keep `InstanceInfo` in the `internal` package?